test/e2e/common.go

```diff
@@ -257,9 +259,25 @@ func EnsureControlPlaneInitialized(ctx context.Context, input clusterctl.ApplyCl
 		Namespace: cluster.Spec.ControlPlaneRef.Namespace,
 		Name:      cluster.Spec.ControlPlaneRef.Name,
 	}
+
+	By("Ensuring KubeadmControlPlane is initialized")
+	Eventually(func() error {
+		if err := getter.Get(ctx, key, kubeadmControlPlane); err != nil {
+			return err
```

Comment:
I kinda think this is a candidate for passing in Gomega into the anonymous func inside `Eventually` so that we can do assertions against the errs and have more meaningful diagnostic stdout.